integ/components/deadline/deadline_01_repository/lib/repository-testing-tier.ts

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as path from 'path';
+import { ILogGroup } from '@aws-cdk/aws-logs';
+import { CfnOutput, Construct } from '@aws-cdk/core';
+import { MongoDbInstaller, MongoDbSsplLicenseAcceptance, MongoDbVersion } from 'aws-rfdk';
+import { Repository } from 'aws-rfdk/deadline';
+import { StorageStruct } from '../../../../lib/storage-struct';
+import { TestingTier, TestingTierProps } from '../../../../lib/testing-tier';
+
+/**
+ * Interface for RepositoryTestingTier properties
+ */
+export interface RepositoryTestingTierProps extends TestingTierProps {
+  /**
+   * Array of StorageStructs representing different test cases
+   */
+  readonly structs: Array<StorageStruct>;
+}
+
+/**
+ * Testing Tier for the Deadline Repository integration test
+ *
+ * Creates a test bastion and configures it to connect to one or more Deadline Repository constructs for testing
+ *
+ * Resources Deployed
+ * ------------------------
+ * - A BastionLinuxHost instance
+ *
+ * Security Considerations
+ * ------------------------
+ * - The bastion instance created by this test is configured to access farm resources on their default ports
+ *   Test scripts stored on the bastion are used to read/write to the repository database and the file system.
+ *   At execution the tests retrieve the value of secrets for the database password and authentication cert.
+ */
+export class RepositoryTestingTier extends TestingTier {
+  constructor(scope: Construct, id: string, props: RepositoryTestingTierProps) {
+    super(scope, id, props);
+
+    const structs = props.structs;
+    structs.forEach( storageStruct => {
+
+      const testSuiteId = 'DL' + (structs.indexOf(storageStruct) + 1).toString();
+
+      const repo = storageStruct.repo;
+      this.configureRepo(testSuiteId, repo);
+
+      const database = storageStruct.database;
+      this.configureDatabase(testSuiteId, database);
+
+      const efs = storageStruct.efs;
+      this.testInstance.connections.allowToDefaultPort(efs);
+
+      const cert = storageStruct.database.cert;
+      this.configureCert(testSuiteId, cert);
+    });
+
+    this.configureBastionUserData({
+      testingScriptPath: path.join(__dirname, '../scripts/bastion/testing'),
+    });
+    this.fetchDocdbCert();
+    this.installMongodb();
+
+  }
+
+  /**
+   * Mounts the Deadline repository's file system to the bastion and outputs the name of its log group
+   *
+   * @param testSuiteId Test case to configure the repository for
+   * @param repo Repository object to connect to the test Bastion
+   */
+  private configureRepo(testSuiteId:string, repo:Repository) {
+    const logGroup = repo.node.findChild('RepositoryLogGroup') as ILogGroup;
```

Comment:
Is it guaranteed that `repo` will have a child `RepositoryLogGroup`?
In other places I've noticed that we often use `tryFindChild` and the check the result:
```
const logGroup = repo.node.tryFindChild(RepositoryLogGroup) as ILogGroup);
if (logGroup)
```

Reply:
It should definitely always exist but for better resilience/error-handling I've fixed this to use the convention you outlined, and also fixed two other places in `storage-struct` where `FindChild` is used similarly.

Reply:
If it's guaranteed to always exist, then I'm fine with the original implementation.
And if we use `tryFindChild`, we should probably add some error reporting, otherwise we now silently skip an issue (although it's unlikely to happen):
```
if (!logGroup) {
   throw new Error(...);
}
```

Reply:
By my read of the code it definitely should always be present, but there's a small chance that the name might not match up if something weird is happening. I've added the error handling so it will hopefully be obvious if something happens to break the function.